### PR TITLE
fix: security restgaps

### PR DESCRIPTION
**Pull request by Hardikl**
security_account_labels [svm].   --> Done

security_certificate_labels [svm].  --> Done

security_labels [uuid] --> This is special case where we have added dummy zapi call and uuid is the instance_key in zapi.

support_labels [node]. --> REST don't have node field but zapi would need as instance_key
image

metric: security [security_audit_destination_status] --> REST don't have data for this.
image

hardikl@hardikl-mac-0 harvest % bin/harvest doctor diffzapirest --zapidatacenter zapi --restdatacenter rest
################## Label value Diffs in prometheus ##############
******* Mismatch in label length. Check data snapmirror_labels source_node
******* Mismatch in label length. Check data snapmirror_labels last_transfer_type
******* Mismatch in label length. Check data snapmirror_labels schedule
******* Mismatch in label length. Check data snapmirror_labels unhealthy_reason
******* Mismatch in label length. Check data qtree_labels qtree
******* Mismatch in label length. Check data node_labels location
******* Mismatch in label length. Check data volume_labels aggr
******* Mismatch in label length. Check data volume_labels node
******* Mismatch in label length. Check data volume_labels snapshot_policy
******* Mismatch in label length. Check data volume_labels all_sm_healthy
################## Missing Labels ##############
shelf_labels []
snapmirror_labels []
security_account_labels []
ntpserver_labels []
qtree_labels []
security_login_labels []
disk_labels []
node_labels []
support_labels [node]
security_labels [uuid]
volume_labels []
lun_labels []
svm_labels []
aggr_labels []
################## Missing Metrics by Object in prometheus ##############
aggr [aggr_inode_files_total aggr_inode_inodefile_public_capacity aggr_inode_files_used aggr_inode_maxfiles_possible aggr_inode_files_private_used aggr_inode_maxfiles_used aggr_inode_maxfiles_available aggr_hybrid_cache_size_total aggr_inode_inodefile_private_capacity aggr_space_physical_used_percent aggr_inode_used_percent aggr_snapshot_inode_used_percent aggr_space_reserved]
qtree [qtree_threshold]
security [security_audit_destination_status]
################## Missing Metrics from dashboard ##############
security [security_audit_destination_status]
aggr [aggr_inode_files_total aggr_inode_inodefile_public_capacity aggr_inode_files_used aggr_inode_inodefile_private_capacity aggr_inode_used_percent aggr_snapshot_inode_used_percent]
##################
hardikl@hardikl-mac-0 harvest %

### CODE REVIEW

**Comment by Hardikl** [comment]
> 

The same cluster(infinity) don't have response in zapi as well. 
image

Need to check how this metric is populated
image

**Comment by Hardikl** [comment]
Made the changes and validated them against mobility cluster where the records are exist.
image

image


**diffzapirest with infinity cluster**
hardikl@hardikl-mac-0 harvest % bin/harvest doctor diffzapirest --zapidatacenter zapi --restdatacenter rest
################## Label value Diffs in prometheus ##############
******* Mismatch in label length. Check data volume_labels aggr
******* Mismatch in label length. Check data volume_labels node
******* Mismatch in label length. Check data volume_labels snapshot_policy
******* Mismatch in label length. Check data volume_labels all_sm_healthy
******* Mismatch in label length. Check data qtree_labels qtree
******* Mismatch in label length. Check data snapmirror_labels source_node
******* Mismatch in label length. Check data snapmirror_labels last_transfer_type
******* Mismatch in label length. Check data snapmirror_labels schedule
******* Mismatch in label length. Check data snapmirror_labels unhealthy_reason
******* Mismatch in label length. Check data node_labels location
################## Missing Labels ##############
security_labels [uuid]
qtree_labels []
security_login_labels []
disk_labels []
ntpserver_labels []
shelf_labels []
svm_labels []
security_certificate_labels []
security_account_labels []
snapmirror_labels []
volume_labels []
aggr_labels []
lun_labels []
support_labels [node]
node_labels []
################## Metrics Diffs in prometheus ##############
################## Missing Metrics by Object in prometheus ##############
aggr [aggr_inode_maxfiles_available aggr_inode_maxfiles_used aggr_inode_maxfiles_possible aggr_inode_inodefile_public_capacity aggr_inode_inodefile_private_capacity aggr_space_reserved aggr_hybrid_cache_size_total aggr_inode_files_private_used aggr_inode_used_percent aggr_inode_files_total aggr_inode_files_used aggr_space_physical_used_percent aggr_snapshot_inode_used_percent]
qtree [qtree_threshold]
################## Missing Metrics from dashboard ##############
aggr [aggr_inode_inodefile_public_capacity aggr_inode_inodefile_private_capacity aggr_inode_used_percent aggr_inode_files_total aggr_inode_files_used aggr_snapshot_inode_used_percent]
##################
hardikl@hardikl-mac-0 harvest %

File: conf/rest/9.12.0/security_audit_dest.yaml [hunk 1]
```diff
@@ -4,7 +4,9 @@ query:            api/security/audit/destinations
 object:           security_audit_destination
 
 counters:
-  - ^protocol     => protocol
+  - ^^protocol                   => protocol
+  - ^^address                    => destination
```
Comment (cgrinds) [inline conf/rest/9.12.0/security_audit_dest.yaml:8]: ^^ == instance key but not listed in `instance_keys` section below?

File: conf/zapi/cdot/9.8.0/security_audit_dest.yaml [hunk 2]
```diff
@@ -5,9 +5,11 @@ object:     security_audit_destination
 
 counters:
   cluster-log-forward-info:
-    - ^protocol                  => protocol
-    - port                       => port
+    - ^^protocol                  => protocol
+    - ^^destination               => destination
```
Comment (cgrinds) [inline conf/zapi/cdot/9.8.0/security_audit_dest.yaml:9]: same comment as above

File: conf/rest/9.12.0/security_audit_dest.yaml (hunk 1 above)
Comment (Hardikl) [reply conf/rest/9.12.0/security_audit_dest.yaml]: `only_cluster_instance` flag rest changes will be done later, reverted those changes.

**Review by cgrinds — APPROVED** [review]

**Pull request merged**